### PR TITLE
[python] out-of-order response handling and tcurl support for multiple requests

### PR DESCRIPTION
**Pull request by blampe**
* Allow `tcurl.py` to issue multiple requests on the same connection (e.g., `-d {1..10000}`).
* Add `profile` option to `tcurl.py`.
* Stop creating new connections for every request in `add_peer`.
* Keep track of pending responses so that we can receive them out of order on that single connection.
* Add a `message_id` option to `frame_and_write` so we can respond with the correct id.

@uber/dsg

### CODE REVIEW

File: python/tchannel/tcurl.py [hunk 1]
```diff
@@ -13,10 +17,46 @@
...
 
+    futures = [
+        tcurl(client, hostport, header, body)
+        for hostport, header, body
+        in getattr(itertools, 'izip', zip)(hostports, headers, bodies)
+    ]
+
+    start = time.time()
+
+    profiler = cProfile.Profile()
```
Comment (breerly) [inline python/tchannel/tcurl.py]: Seems like you shouldn't incur the cost of instantiating this is `profile` is `None`.

File: python/tchannel/tcurl.py [hunk 2]
```diff
@@ -13,10 +17,46 @@
...
+
+    if profile:
+        profiler.disable()
+        profiler.create_stats()
+        stats = pstats.Stats(profiler)
+        stats.strip_dirs().sort_stats('tot').print_stats(15)
+
+    stop = time.time()
+
+    log.debug(
```
Comment (breerly) [inline python/tchannel/tcurl.py]: Nice.

File: python/tchannel/tcurl.py [hunk 3]
```diff
@@ -81,7 +124,43 @@ def parse_args():
         action="store_true"
     )
 
-    return parser.parse_args()
+    parser.add_argument(
```
Comment (breerly) [inline python/tchannel/tcurl.py]: Suggest moving all `parser.add_argument` calls to top of file.

**Comment by breerly** [comment]
Had some nits. Still - :shipit:

File: python/tchannel/tcurl.py (hunk 1 above)
Comment (blampe) [inline python/tchannel/tcurl.py]: Fixed.

File: python/tchannel/tcurl.py (hunk 3 above)
Comment (blampe) [inline python/tchannel/tcurl.py]: Word, done.

**Pull request merged**